game/world/respawnobject.cpp

```diff
@@ -0,0 +1,189 @@
+#include "respawnobject.h"
+
+#include <Tempest/Log>
+
+#include "game/serialize.h"
+#include "world/world.h"
+#include "gothic.h"
+
+#include <vector>
+#include <map>
+
+using namespace Tempest;
+
+std::vector<RespawnObject> RespawnObject::respawnList = {};
+std::string RespawnObject::processedDayWorld = "";
+uint32_t RespawnObject::minDays = 2;  // Should be at least 1
+uint32_t RespawnObject::maxDays = 10; // Must be bigger than minDays
+
+RespawnObject::RespawnObject(Serialize &fin){
+    fin.read(inst,wp,respawnDay,world);
+}
+
+void RespawnObject::save(Serialize &fout) const{
+    fout.write(inst,wp,respawnDay,world);
+}
+
+void RespawnObject::registerObject(size_t inst, std::string wp, uint32_t guild){
+    if (!RespawnObject::npcShouldRespawn(guild))
+        return;
+    World *world = Gothic::inst().world();
+    // Skip if waypoint is unknown
+    if (!world->findPoint(wp))
+        return;
+    RespawnObject ro;
+    ro.inst = inst;
+    ro.wp = wp;
+    uint32_t currDay = getCurrDay(world);
+    uint32_t offset = minDays + world->script().rand(maxDays - minDays);
+    ro.respawnDay = currDay + offset;
+    ro.world = world->name();
+
+    printMsg("Register respawn for day " + std::to_string(ro.respawnDay) + " (" + ro.world + ")");
+
+    // Store for respawn in respawn list
+    RespawnObject::respawnList.push_back(ro);
+}
+
+void RespawnObject::processRespawnList(){
+    // Is respawn activated ?
+    if (!Gothic::inst().doRespawn())
+        return;
+    World* world  = Gothic::inst().world();
+    // Ensure respawn list is not processed during world transition / loading sequences
+    if(world==nullptr || Gothic::inst().player()==nullptr) {
+        printMsg("Skip processing respawn list (world not fully loaded)");
+        return;
+    }
+    uint32_t currDay = getCurrDay(world);
+    // Check against cache value for early abort (no multi-processing on same day in same world)
+    std::string currDayWld = std::to_string(currDay) + std::string("_") + std::string(world->name());
+    if (processedDayWorld.compare(currDayWld) == 0) {
```

Comment:
Case sensitivity? As touched in https://github.com/Try/OpenGothic/issues/349 mod may have mixed case in world name and get away with it. Original game seems using regular windows-files to store per-world state